### PR TITLE
Update installation instructions

### LINKED ISSUE #3107: List supported architectures / hardware / OS in docs
Seems like there's no list of supported hardware, architectures, GPU types, Operating systems etc in the docs/GH pages. This ticket is to add it! (idea from private IRC with @AndiH)

### PR DESCRIPTION
**Pull request by seibert**
This cleans up the installation instructions, and gives information for all of the other platforms Numba now supports.

Marked as [WIP] until there are armv7 packages in the Numba channel and we get the Jetconda installer sorted out.

Fixes #3107.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -45,76 +39,12 @@ Distribution: https://www.anaconda.com/download
...
-in order to override the standard search paths:
-
-NUMBAPRO_CUDA_DRIVER
-  Path to the CUDA driver shared library
-NUMBAPRO_NVVM
-  Path to the CUDA libNVVM shared library file
-NUMBAPRO_LIBDEVICE
-  Path to the CUDA libNVVM libdevice directory which contains .bc files
-
+For more options, see the Installation Guide: http://numba.pydata.org/numba-doc/latest/user/installing.html
```
Comment (stuartarchibald) [inline README.rst]: Couple of lines above this there's a `numba` in `The easiest way to install numba and ` that should have a capital `N`

File: README.rst [hunk 2]
```diff
@@ -45,76 +39,12 @@ Distribution: https://www.anaconda.com/download
...
 
 Documentation
 =============
 
-http://numba.pydata.org/numba-doc/dev/index.html
+http://numba.pydata.org/numba-doc/latest/index.html
```
Comment (stuartarchibald) [inline README.rst]: 3 or 4 lines below... Think Gmane doesn't track the list any more?

File: README.rst [hunk 3]
```diff
@@ -129,13 +59,6 @@ http://news.gmane.org/gmane.comp.python.numba.user
...
 
-Website
-=======
-
-See if our sponsor can help you (which can help this project): https://www.anaconda.com
-
-http://numba.pydata.org
-
 
 Continuous Integration
```
Comment (stuartarchibald) [inline README.rst]: This ought to be just badges? Also Appveyor is missing?

File: README.rst [hunk 4]
```diff
@@ -6,26 +6,20 @@ Numba
...
+Numba can compile a large subset of numerically-focused Python, including many
+NumPy functions.  Additionally, Numba has support for automatic
+parallelization of loops, generation of GPU-accelerated code, and creation of
+ufuncs and C callbacks.
 
+For more information about Numba, see the Numba homepage: 
+http://numba.pydata.org
 
 Dependencies
 ============
```
Comment (stuartarchibald) [inline README.rst]: There's a `numpy` that should be `NumPy` a couple of lines below.

File: README.rst [hunk 5]
```diff
@@ -6,26 +6,20 @@ Numba
    :target: https://gitter.im/numba/numba?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge
    :alt: Gitter
```
Comment (stuartarchibald) [inline README.rst]: Put the logo here?

File: docs/source/developer/contributing.rst [hunk 6]
```diff
@@ -80,51 +80,54 @@ case, please prepend ``[WIP]`` to your pull request's title.
 Build environment
 '''''''''''''''''
 
-Numba has a number of dependencies (mostly `Numpy <http://www.numpy.org/>`_
```
Comment (stuartarchibald) [inline docs/source/developer/contributing.rst]: Just noticed that 2-3 lines above this there's a `platorm` which would benefit from an 'f'.

File: docs/source/developer/contributing.rst [hunk 7]
```diff
@@ -80,51 +80,54 @@ case, please prepend ``[WIP]`` to your pull request's title.
...
+Numba has a number of dependencies (mostly `numpy <http://www.numpy.org/>`_
```
Comment (stuartarchibald) [inline docs/source/developer/contributing.rst]: Isn't it `NumPy`?

File: docs/source/developer/contributing.rst [hunk 8]
```diff
@@ -80,51 +80,54 @@ case, please prepend ``[WIP]`` to your pull request's title.
...
-the llvmlite library::
+First add the Anaconda Cloud ``numba`` channel so as to get development builds
+of the llvmlite library::
 
    $ conda config --add channels numba
 
 Then create an environment with the right dependencies::
 
-   $ <path_to_miniconda>/conda create -n numbaenv python=3.5 llvmlite numpy
+   $ conda create -n numbaenv python=3.6 llvmlite numpy scipy
```
Comment (stuartarchibald) [inline docs/source/developer/contributing.rst]: To get full feature support, IIRC `scipy`, `ipython`, `jinja2` and `cffi` are needed. Also once the thread layer stuff goes in `tbb` and on OSX `llvm-openmp` and `intel-openmp`. For AOT and compiling the source a working C/C++ compiler is needed. These are available from compiler packages like `gcc_linux-64` and `gxx_linux-64` on linux and `clang_osx-64` and `clangxx_osx-64` on OSX, windows needs an appropriate MSVC version.

File: docs/source/developer/contributing.rst [hunk 9]
```diff
@@ -80,51 +80,54 @@ case, please prepend ``[WIP]`` to your pull request's title.
...
 ''''''''''''''
 
 For a convenient development workflow, we recommend you build Numba inside
 its source checkout::
 
+   $ git clone git://github.com/numba/numba.git
+   $ cd numba
    $ python setup.py build_ext --inplace
 
 This assumes you have a working C compiler and runtime on your development
```
Comment (stuartarchibald) [inline docs/source/developer/contributing.rst]: The thing 2-3 lines below about re-running that command when C files are mutated... it doesn't always work.

File: docs/source/developer/contributing.rst [hunk 10]
```diff
@@ -217,27 +220,22 @@ also needs to pass the test suite before it is merged in.
...
-Each time you submit a pull request, a corresponding build will be started
-at Travis-CI and check that Numba builds and tests without any errors.
-You can expect this to take less than 20 minutes.
-
-Some platforms (such as Windows) cannot be hosted by Travis-CI, and the
-Numba team has therefore access to a separate platform provided by
-`Anaconda, Inc. <https://anaconda.com>`_, our sponsor. We hope parts of that
-infrastructure can be made public in the future.
+Every commit to the master branch is automatically tested on all of the
+platform Numba supports.  This includes ARMv7, ARMv8, POWER8, as well as both
```
Comment (stuartarchibald) [inline docs/source/developer/contributing.rst]: s/platform/platforms/

File: docs/source/developer/contributing.rst [hunk 11]
```diff
@@ -217,27 +220,22 @@ also needs to pass the test suite before it is merged in.
...
+AMD and NVIDIA GPUs.  The build system however is internal to Anaconda, so we
+also use `Travis CI <https://travis-ci.org/numba/numba>`_ to provide public
```
Comment (stuartarchibald) [inline docs/source/developer/contributing.rst]: and Appveyor?

File: docs/source/user/installing.rst [hunk 12]
```diff
@@ -17,60 +17,192 @@ Our supported platforms are:
...
 or::
 
-   $ conda update numba
+    $ conda update numba
+
+Note that Numba, like Anaconda, only supports PPC in 64-bit little-endian mode.
+
+To enable CUDA GPU support for Numba, install the latest `graphics drivers from
+NVIDIA <https://www.nvidia.com/Download/index.aspx>`_ for your platform.
+(Note that the open source Nouveau drivers shipped by default with Ubuntu do
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: Think most Linux's ship with Nouveau by default.

File: docs/source/user/installing.rst [hunk 13]
```diff
@@ -17,60 +17,192 @@ Our supported platforms are:
...
+Installing using pip on x86/x86_64 Platforms
+--------------------------------------------
+
+Binary wheels for Windows, Mac, and Linux are also available from `PyPI
+<https://pypi.org/project/numba/>`_.  You can install Numba using ``pip``::
+
+    $ pip install numba
+
+This will download all of the needed dependencies as well.  You do not need to
+have *LLVM* installed to use Numba (in fact, Numba will ignore all LLVM
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: Why italicize LLVM ?

File: docs/source/user/installing.rst [hunk 14]
```diff
@@ -17,60 +17,192 @@ Our supported platforms are:
...
+versions installed on the system) as the required components are bundled into
+the llvmlite wheel.
+
+To use CUDA with Numba installed by `pip`, you need to install the `CUDA SDK
+<https://developer.nvidia.com/cuda-downloads>`_ from NVIDIA.  Then you may need to
+set the following environment variables so Numba can locate the required
+libraries:
+
+  * ``NUMBAPRO_CUDA_DRIVER`` - Path to the CUDA driver shared library file
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: Due to the leading indent here, it renders slightly weird, in a block and with bold fonts.

File: docs/source/user/installing.rst [hunk 15]
```diff
@@ -17,60 +17,192 @@ Our supported platforms are:
...
+---------------------------------------------
+
+Although many ARMv8 platforms are likely to require building Numba from
+source, we do build and test conda packages on the `NVIDIA Jetson TX2
+<https://www.nvidia.com/en-us/autonomous-machines/embedded-systems-dev-kits-modules/>`_.
+These packages rely on an infrequently updated port of Berryconda to ARMv8
+called `Jetconda <https://github.com/seibert/jetconda/releases>`_.  Jetconda
+is built on Ubuntu 16.04 running on the Jetson TX2, although the packages may
+work on other ARMv8 systems running Ubuntu 16.04.  Once Jetconda is installed,
+Numba can be installed from the ``numba`` channel
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: Double colon `::` after `channel` else the conda command won't end up in a code block.

File: docs/source/user/installing.rst [hunk 16]
```diff
@@ -17,60 +17,192 @@ Our supported platforms are:
...
+
+
+Installing on Linux ARMv7 Platforms
+-----------------------------------
+
+`Berryconda <https://https://github.com/jjhelmus/berryconda>`_ is a
+conda-based Python distribution for the Raspberry Pi.  We are now uploading
+packages to the ``numba`` channel on Anaconda Cloud for ARMv7-based boards,
+which currently incudes the the Raspberry Pi 2 and 3, but not the Pi 1 or
+Zero.  These can be installed using conda from the `numba` channel::
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: Double backticks around `numba` for consistency.

File: docs/source/user/installing.rst [hunk 17]
```diff
@@ -17,60 +17,192 @@ Our supported platforms are:
...
-You will need a C compiler corresponding to your Python installation, as
-well as the `Numpy <http://www.numpy.org/>`_ and
-`llvmlite <https://github.com/numba/llvmlite>`_ packages.  See :ref:`buildenv`
-for more information.
+Source archives of the latest release can also be found on
+`PyPI <https://pypi.org/project/numba/>`_.  In addition to ``llvmlite``, you will also need:
 
-Checking your installation
---------------------------
+  * A C compiler compatible with your Python installation
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: The indenting of the list make it render strangely. Also, want to mention the Anaconda toolchains?

File: docs/source/user/installing.rst [hunk 18]
```diff
@@ -17,60 +17,192 @@ Our supported platforms are:
...
+  * `Numpy <http://www.numpy.org/>`_
```
Comment (stuartarchibald) [inline docs/source/user/installing.rst]: `s/Numpy/NumPy/`

File: README.rst (hunk 5 above)
Comment (seibert) [reply README.rst]: Going to skip this for now, but we can do this in a future PR.

**Comment by seibert** [comment]
I think I've responded to all the comments.  Once Travis CI confirms the documentation builds, I'll merge.

**Comment by seibert** [comment]
ok, doc build passed, merging

**Pull request merged**